flow/project.py

```diff
@@ -4689,7 +4689,9 @@ def _main_status(self, args):
                     "complete the update anyways or '--show-traceback' to show "
                     "the full traceback."
                 )
-                error = error.__cause__  # Always show the user traceback cause.
+                if isinstance(error, (UserOperationError, UserConditionError)):
+                    # Always show the user traceback cause.
+                    error = error.__cause__
```

Comment:
Can we add a test for an invalid user operation? I'm kind of surprised that we don't have one, maybe this missing coverage is erroneous?

Reply:
We don't have a test for this. I think we'd have to create a new test class with a `FlowProject` subclass that raises an error in a condition function, and then assert that some substring is in the error output, which takes some work. That felt messy and prone to breaking so I skipped adding proper tests.

Reply:
I spent 20 minutes trying to mock this -- it's messy. I think I'd need to add a new `define_project_invalid_condition.py` and configure its entrypoint and some other steps just for this test of tracebacks. I think this has to be triggered through the `Project.main()` interface, so a self-contained Python test class won't cut it (unless I'm missing something easier).

Reply:
Why do you think it would be prone to breaking? I think your description of what test would need to be is correct, but that doesn't feel any harder than what you added in #609. You'd just have to define the operation as raise an exception (just `assert False`) and then use try-except in the test, right? We've had multiple recent issues related to exception handling with user operations so I think it's important for us to improve our coverage here if we can.

Reply:
We're probably not taking good advantage of pytest's powerful fixture system at the moment, because the test framework was setup before we switched to pytest. Rather than wasting a lot effort on trying to get this to work, maybe we should consider to refactor the test framewor to use a more module and fixture hierarchy that should make it easier to create these kind of scenarios. What do you think?

Reply:
@bdice  I'll be happy to try refactoring the testing framework!

Reply:
@kidrahahjo I think that would be great, but we should make sure it's a targeted effort. Would you mind doing a brief analysis of what could potentially be improved and then record that in an issue before making any changes?

Reply:
As far as testing this specific case, @bdice your analysis of how you need to write this test is correct, and if you've attempted it and found it challenging I am fine postponing. I agree that any refactoring should be discussed before we move forward. Since the vast majority of signac-flow users employ our CLI, though, I think improving both our test coverage and our ability to test these kind of CLI-only issues is valuable and worth exploring.